core/src/banking_stage/consumer.rs

```diff
@@ -206,6 +206,26 @@ impl Consumer {
             .slot_metrics_tracker
             .increment_retryable_packets_count(retryable_transaction_indexes.len() as u64);
 
+        // Now we track the performance for the interested transactions which is not in the retryable_transaction_indexes
+        // We assume the retryable_transaction_indexes is already sorted.
+        for (index, packet) in packets_to_process.iter().enumerate() {
```

Comment:
I do not see an easy way to do that -- I would like avoid smudge on the SanitizedTransaction. But I think I can make this more efficient -- I do not need the binary search on the retryable index, I could do it in one simple loop.

Reply:
are the actual iterations not like one stack frame deeper in most cases?

Reply:
This probably needs to be done elsewhere; this is never called with the new scheduler, so we'd never get any metrics if that's enabled. By the time this change gets in, it will be the default.

Reply:
@apfitzge what is the new scheduler's name?

Reply:
Talked to @apfitzge offline, will address both issue raised by Trent and Andrew.